cli/popper/gha.py

```diff
@@ -215,8 +219,17 @@ def download_actions(self):
                 pu.info('[popper] cloning actions from repositories\n')
                 infoed = True
 
-            scm.clone(user, repo, repo_parent_dir, version, debug=self.debug)
-
+            repo_dir = os.path.join(repo_parent_dir, repo)
+            if os.path.exists(repo_dir):
+                pu.exec_cmd('rm -rf {}'.format(repo_dir))
```

Comment:
I'm not familiar with `GitPython`, do you know if it is possible to avoid re-cloning a repo if it is already cloned? This would allow us to avoid having to `rm -rf` which is always scary.

Reply:
@ivotron what we can do is check if the repo is present locally , we avoid cloning . Thus getting rid of `rm -rf`

Reply:
yes, hopefully that can be handled by gitpython. The main issue here is, how do we ensure that the local folder corresponds to the repo we want to clone? E.g. two repos might have the same name but they are distinct. So we might need to create a full `/tmp/actions/github/org/repo` path (as opposed to what we do know which is to put it in `/tmp/actions/repo`.

Reply:
hey @JayjeetAtGithub, thanks a lot for working on this. Looks like my comment above was not addressed, right? No biggie if not, we can open another issue for that. I just wanted to double-check that it will not be part of this PR, so I can proceed to merge it. Thanks!

Reply:
@ivotron I am working on this part currently . Updating soon

Reply:
@ivotron is `/github/` important in the path or we can do `/tmp/actions/org/repo` ?

Reply:
@ivotron I guess the url can be `https://github.com` or `https://gitlab.com` etc. So if this is the case, can the path be like `/tmp/actions/gitlab.com/org/repo` ?

Reply:
Can this be done by comparing the remote origins of the repos? if they are same i avoid cloning else i remove the exisiting one and clone

Reply:
@ivotron i feel it would be better if we can solve this `/github/org/repo` issue in another PR as this PR has already lots of changes. I would be better to have a clean PR for this. Thanks a lot !

Reply:
sure. Thanks!